src/connectors/snowflake/trulens/connectors/snowflake/connector.py

```diff
@@ -118,9 +74,80 @@ def __init__(
             database_prefix,
             database_args,
             database_check_revision,
-            connection_parameters,
+            URL(**connection_parameters),
         )
 
+    def _create_snowpark_session(
+        self, connection_parameters: Dict[str, Optional[str]]
+    ):
+        connection_parameters = connection_parameters.copy()
+        # Validate.
+        connection_parameters_to_set = []
+        for k, v in connection_parameters.items():
+            if v is None:
+                connection_parameters_to_set.append(k)
+        if connection_parameters_to_set:
+            raise ValueError(
+                f"If not supplying `snowpark_session` then must set `{connection_parameters_to_set}`!"
+            )
+        self.password_known = True
+        # Create snowpark session making sure to create schema if it doesn't
+        # already exist.
+        schema = connection_parameters["schema"]
+        del connection_parameters["schema"]
```

Comment:
just curious - this is done so that you don't have to explicitly catch/except the connection error when the passed in schema from connection params is not valid or not recognized right?

Reply:
yes; since we want to be able to handle the situation where we just create the schema on the fly.